src/components/PopupWithYes.js

```diff
@@ -0,0 +1,26 @@
+import Popup from './Popup.js';
+
+export default class PopupWithYes extends Popup{
+    constructor({popupSelector, handleYesButton}){
+        super({popupSelector});
+        this._handleYesButton = handleYesButton;
+        this._popupButton = this._popupElement.querySelector('.modal__button')
+    }
+
+    setEventListeners(){
+        super.setEventListeners();
+
+        this._popupElement.querySelector('.modal__close-button-container').addEventListener('click', () => {
+            this._popupButton.removeEventListener('click', deleteCard);
```

Comment:
**Could be improved**

You can update the function like that (without any removing it 

```js

  open(deleteCard) {
    super.open();
    this._deleteCard = deleteCard
  }
```